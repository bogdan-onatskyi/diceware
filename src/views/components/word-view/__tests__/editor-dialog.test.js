import React from 'react';

import EditorDialog from '../editor-dialog';
import WordView from '../word-view';
import WordObject from '../../../../models/word-object';

describe("EditorDialog component", () => {
    const wordObject = new WordObject(1);

    const setup = ({
                       toggleEditor = wordObject.toggleEditor,
                       handleFilter = wordObject.handleFilter,
                       handleCountWords = wordObject.handleCountWords,
                       filter = wordObject.filter,
                       handleBackspace = wordObject.handleBackspace,
                       wordView = <WordView/>
                   }) => {

        const wrapper = shallow(<EditorDialog toggleEditor={toggleEditor}
                                              handleFilter={handleFilter}
                                              handleCountWords={handleCountWords}
                                              filter={filter}
                                              handleBackspace={handleBackspace}
                                              wordView={wordView}/>);
        return {wrapper};
    };

    beforeEach(() => {
        wordObject.filter = '';
    });

    it("should match saved snapshot", () => {
        const {wrapper} = setup({});
        expect(wrapper).toMatchSnapshot();
    });


    it("should call handleFilter with a char when a char button is pressed", () => {
        const handleFilter = spyOn(wordObject, 'handleFilter').and.callThrough();
        const {wrapper} = setup({handleFilter: handleFilter});
        const alfabet = 'qwertyuiopasdfghjklzxcvbnm'.split('');

        const charButtons = wrapper.find({type: 'char'});
        charButtons.forEach((button, i) => {
            button.simulate('click');
            expect(handleFilter).toHaveBeenCalledWith(alfabet[i]);
        });
    });

    it("should call handleBackspace when the back-space button is pressed", () => {
        wordObject.filter = 'any';
        const handleBackspace = spyOn(wordObject, 'handleBackspace').and.callThrough();
        const {wrapper} = setup({handleBackspace: handleBackspace});

        wrapper.find({type: 'back-space'}).simulate('click');
        expect(handleBackspace).toHaveBeenCalled();
        expect(wordObject.filter).toEqual('an');
    });

    it("should have 6 props", () => {
        const propsStr = 'toggleEditor handleFilter handleCountWords filter handleBackspace wordView';
        const props = [];
        for (let prop in EditorDialog.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});
