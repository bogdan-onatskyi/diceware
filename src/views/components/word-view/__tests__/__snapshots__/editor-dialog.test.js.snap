// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorDialog component should match saved snapshot 1`] = `
<ModalDialog
  handlerClose={[Function]}
  isOpen={true}
  title="Редактор слова"
>
  <WordView />
  <div
    key="row_0"
  >
    <Button
      data-tip="Добавить букву <Q> к фильтру"
      disabled={false}
      key="char_q"
      onClick={[Function]}
      type="char"
    >
      q
    </Button>
    <Button
      data-tip="Добавить букву <W> к фильтру"
      disabled={false}
      key="char_w"
      onClick={[Function]}
      type="char"
    >
      w
    </Button>
    <Button
      data-tip="Добавить букву <E> к фильтру"
      disabled={false}
      key="char_e"
      onClick={[Function]}
      type="char"
    >
      e
    </Button>
    <Button
      data-tip="Добавить букву <R> к фильтру"
      disabled={false}
      key="char_r"
      onClick={[Function]}
      type="char"
    >
      r
    </Button>
    <Button
      data-tip="Добавить букву <T> к фильтру"
      disabled={false}
      key="char_t"
      onClick={[Function]}
      type="char"
    >
      t
    </Button>
    <Button
      data-tip="Добавить букву <Y> к фильтру"
      disabled={false}
      key="char_y"
      onClick={[Function]}
      type="char"
    >
      y
    </Button>
    <Button
      data-tip="Добавить букву <U> к фильтру"
      disabled={false}
      key="char_u"
      onClick={[Function]}
      type="char"
    >
      u
    </Button>
    <Button
      data-tip="Добавить букву <I> к фильтру"
      disabled={false}
      key="char_i"
      onClick={[Function]}
      type="char"
    >
      i
    </Button>
    <Button
      data-tip="Добавить букву <O> к фильтру"
      disabled={false}
      key="char_o"
      onClick={[Function]}
      type="char"
    >
      o
    </Button>
    <Button
      data-tip="Добавить букву <P> к фильтру"
      disabled={false}
      key="char_p"
      onClick={[Function]}
      type="char"
    >
      p
    </Button>
  </div>
  <div
    key="row_1"
  >
    <Button
      data-tip="Добавить букву <A> к фильтру"
      disabled={false}
      key="char_a"
      onClick={[Function]}
      type="char"
    >
      a
    </Button>
    <Button
      data-tip="Добавить букву <S> к фильтру"
      disabled={false}
      key="char_s"
      onClick={[Function]}
      type="char"
    >
      s
    </Button>
    <Button
      data-tip="Добавить букву <D> к фильтру"
      disabled={false}
      key="char_d"
      onClick={[Function]}
      type="char"
    >
      d
    </Button>
    <Button
      data-tip="Добавить букву <F> к фильтру"
      disabled={false}
      key="char_f"
      onClick={[Function]}
      type="char"
    >
      f
    </Button>
    <Button
      data-tip="Добавить букву <G> к фильтру"
      disabled={false}
      key="char_g"
      onClick={[Function]}
      type="char"
    >
      g
    </Button>
    <Button
      data-tip="Добавить букву <H> к фильтру"
      disabled={false}
      key="char_h"
      onClick={[Function]}
      type="char"
    >
      h
    </Button>
    <Button
      data-tip="Добавить букву <J> к фильтру"
      disabled={false}
      key="char_j"
      onClick={[Function]}
      type="char"
    >
      j
    </Button>
    <Button
      data-tip="Добавить букву <K> к фильтру"
      disabled={false}
      key="char_k"
      onClick={[Function]}
      type="char"
    >
      k
    </Button>
    <Button
      data-tip="Добавить букву <L> к фильтру"
      disabled={false}
      key="char_l"
      onClick={[Function]}
      type="char"
    >
      l
    </Button>
  </div>
  <div
    key="row_2"
  >
    <Button
      data-tip="Добавить букву <Z> к фильтру"
      disabled={false}
      key="char_z"
      onClick={[Function]}
      type="char"
    >
      z
    </Button>
    <Button
      data-tip="Добавить букву <X> к фильтру"
      disabled={false}
      key="char_x"
      onClick={[Function]}
      type="char"
    >
      x
    </Button>
    <Button
      data-tip="Добавить букву <C> к фильтру"
      disabled={false}
      key="char_c"
      onClick={[Function]}
      type="char"
    >
      c
    </Button>
    <Button
      data-tip="Добавить букву <V> к фильтру"
      disabled={false}
      key="char_v"
      onClick={[Function]}
      type="char"
    >
      v
    </Button>
    <Button
      data-tip="Добавить букву <B> к фильтру"
      disabled={false}
      key="char_b"
      onClick={[Function]}
      type="char"
    >
      b
    </Button>
    <Button
      data-tip="Добавить букву <N> к фильтру"
      disabled={false}
      key="char_n"
      onClick={[Function]}
      type="char"
    >
      n
    </Button>
    <Button
      data-tip="Добавить букву <M> к фильтру"
      disabled={false}
      key="char_m"
      onClick={[Function]}
      type="char"
    >
      m
    </Button>
  </div>
  <div>
    <Button
      data-tip=""
      disabled={true}
      onClick={[Function]}
      type="back-space"
    >
      backspace
    </Button>
  </div>
</ModalDialog>
`;
